src/macaron/__main__.py

```diff
@@ -75,58 +77,45 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
+def verify_policy(action_args: argparse.Namespace) -> int:
+    """Run policy engine and verify the Datalog policy.
 
-    if not policy_files:
-        logger.error("The policy is not provided to complete this action.")
-        sys.exit(os.EX_NOINPUT)
+    Returns
+    -------
+    int
+        Returns os.EX_OK if successful or the corresponding error code on failure.
+    """
+    if not os.path.isfile(action_args.database):
+        logger.critical("The database file does not exist.")
+        return os.EX_OSFILE
 
-    if len(policy_files) > 1:
-        logger.error("Exactly one policy must be provided to complete this action.")
-        sys.exit(os.EX_USAGE)
+    result = run_policy_engine(action_args.database, action_args.show_prelude, action_args.file)
+    policy_reporter = PolicyReporter()
+    policy_reporter.generate(global_config.output_path, result)
 
-    policy_file = policy_files[0]
-    try:
-        prov_content = ProvPayloadLoader.load(prov_file)
-        policy: Policy | None = Policy.make_policy(policy_file)
+    if ("failed_policies" in result) and any(result["failed_policies"]):
+        return os.EX_DATAERR
 
-        if not policy:
-            logger.error("Could not load policy at %s.", policy_file)
-            sys.exit(os.EX_NOINPUT)
-
-        logger.info("Validating the provenance at %s against %s.", prov_file, policy)
-
-        if not policy.validate(prov_content):
-            logger.error("The validation for provenance at %s was unsuccessful.", prov_file)
-            sys.exit(os.EX_NOINPUT)
-
-        logger.info("The validation for provenance at %s was successful.", prov_file)
-        sys.exit(os.EX_OK)
-    except (SLSAProvenanceError, PolicyRuntimeError) as error:
-        logger.error(error)
-        sys.exit(os.EX_DATAERR)
+    return os.EX_OK
 
 
 def perform_action(action_args: argparse.Namespace) -> None:
     """Perform the indicated action of Macaron."""
-    if action_args.action == "dump_defaults":
-        # Create the defaults.ini file in the output dir and exit.
-        create_defaults(action_args.output_dir, os.getcwd())
-        sys.exit(os.EX_OK)
-
-    # Check that the GitHub token is enabled.
-    if not action_args.personal_access_token:
-        logger.error("GitHub access token not set.")
-        sys.exit(os.EX_USAGE)
-
     match action_args.action:
+        case "dump-defaults":
+            # Create the defaults.ini file in the output dir and exit.
+            create_defaults(action_args.output_dir, os.getcwd())
+            sys.exit(os.EX_OK)
+
+        case "verify-policy":
+            sys.exit(verify_policy(action_args))
+
         case "analyze":
+            # Check that the GitHub token is enabled.
+            if not action_args.personal_access_token:
+                logger.error("GitHub access token not set.")
+                sys.exit(os.EX_USAGE)
             analyze_slsa_levels_single(action_args)
```

Comment:
> I think we need to have a separate PR to update `analyze_slsa_levels_single` so that it return the status code instead of handling `os.exit` by itself.

There is already a [ticket](https://github.com/oracle-samples/macaron/issues/175) where this can be handled.

> In additions, do you think that the check for existence of the access token could be relocated inside `analyze_slsa_levels_single` ?

There is already a [ticket](https://github.com/oracle-samples/macaron/issues/199) for this.